src/timeout.py

```diff
@@ -20,44 +20,44 @@
 from logger import Logger
 
 
-class SetTimeout:
+class Timeout:
 
     def __init__(self, configs, args):
         self._db_config = configs['db_configs'][args.db_config]
         api_token = os.getenv('API_TOKEN')
         self._db = kernelci.db.get_db(self._db_config, api_token)
-        self._logger = Logger("config/logger.conf", "set_timeout")
+        self._logger = Logger("config/logger.conf", "timeout")
         self._poll_period = args.poll_period
 
-    def _update_pending_child(self, parent_id):
-        """Set child node status to timeout when parent is timed out"""
-        child_nodes = self._db.get_nodes({"status": "pending",
-                                          "parent": parent_id})
+    def _update_child_nodes(self, parent_id):
+        """Set child node state to done when parent is timed out"""
+        child_nodes = self._db.get_nodes({'parent': parent_id})
         for child in child_nodes:
-            child['status'] = "timeout"
-            self._db.submit({'node': child})
-            self._update_pending_child(child["_id"])
+            if child['state'] != 'done':
+                child['state'] = 'done'
+                self._db.submit({'node': child})
+                self._update_child_nodes(child['_id'])
 
-    def _set_timeout_status(self, node):
-        """Set Node status to timeout if maximum wait time is over"""
+    def _update_closing_node(self, node):
+        """Set Node state to done if maximum wait time is over"""
         current_time = datetime.utcnow()
-        max_wait_time = datetime.fromisoformat(node['created']) + \
-            timedelta(hours=node['max_wait_time'])
-        if current_time > max_wait_time:
-            node['status'] = "timeout"
+        expires = datetime.fromisoformat(node['created']) + \
+            timedelta(hours=node['timeout'])
```

Comment:
I thought the `timeout` field was supposed to be an absolute timestamp rather than a time delta now?  Maybe other parts of the pipeline still need to be updated for that and a follow-up PR will be needed here.

Reply:
This PR is about node state changes. I will create another one for the `timeout` field.